src/modal/modal.ts

```diff
@@ -141,6 +153,68 @@ export class CaptureEyeModal extends LitElement {
     );
   }
 
+  private startEngagementZoneRotation() {
+    if (this._engagementZones.length <= 1) return;
+    this._engagementZoneRotationIntervalId = setInterval(() => {
+      this._engagementZoneIndex =
+        (this._engagementZoneIndex + 1) % this._engagementZones.length;
+    }, this._engagementZoneRotationInterval);
+  }
+
+  private stopEngagementZoneRotation() {
+    if (this._engagementZoneRotationIntervalId)
+      clearInterval(this._engagementZoneRotationIntervalId);
+  }
+
+  private rotateNext() {
+    if (this._engagementZones.length === 0) return;
+    this._engagementZoneIndex =
+      (this._engagementZoneIndex + 1) % this._engagementZones.length;
+  }
+  private rotatePrev() {
+    if (this._engagementZones.length === 0) return;
+    this._engagementZoneIndex =
+      (this._engagementZoneIndex - 1) % this._engagementZones.length;
+  }
+
+  private preloadEngagementZoneImages(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      let loadedImages = 0;
+      const imageUrls =
+        this._engagementZones.length > 0
+          ? this._engagementZones
+              .map((zone) => zone.image)
+              .filter((url) => url !== '')
```

Comment:
maybe the filter can be removed, as the check is already done earlier (line 197 in src/capture-eye.ts).

Reply:
A test passing empty string makes me add this line. I've changed the test since this is not an expected usage and removed this filter.